src/annotator/plugin/test/bucket-bar-test.js

```diff
@@ -35,48 +52,138 @@ describe('BucketBar', () => {
+      document.body.appendChild(containerEl);
+      sandbox.stub(console, 'warn'); // Restored in test-global `afterEach`
+    });
 
-  context('when a bucket is clicked', () => {
-    let bucketBar;
+    afterEach(() => {
+      containerEl.remove();
+    });
 
+    it('will append its element to any supplied `options.container` selector', () => {
+      bucketBar = createBucketBar({ container: '.bucket-bar-container' });
+      assert.exists(containerEl.querySelector('.annotator-bucket-bar'));
+    });
+
+    it('will append itself to the element passed to constructor if `options.container` non-existent', () => {
+      bucketBar = createBucketBar({ container: '.bucket-bar-nope' });
+      assert.notExists(containerEl.querySelector('.annotator-bucket-bar'));
+      assert.calledOnce(console.warn);
+    });
+  });
+
+  describe('updating buckets', () => {
+    it('should update buckets when the window is resized', () => {
+      bucketBar = createBucketBar();
+      assert.notCalled(fakeBucketUtil.buildBuckets);
+      window.dispatchEvent(new Event('resize'));
+      assert.calledOnce(fakeBucketUtil.buildBuckets);
+    });
+
+    it('should update buckets when the window is scrolled', () => {
+      bucketBar = createBucketBar();
+      assert.notCalled(fakeBucketUtil.buildBuckets);
+      window.dispatchEvent(new Event('scroll'));
+      assert.calledOnce(fakeBucketUtil.buildBuckets);
+    });
+
+    context('when scrollables provided', () => {
+      let scrollableEls = [];
+
+      beforeEach(() => {
+        const scrollableEls1 = document.createElement('div');
+        scrollableEls1.className = 'scrollable-1';
+        const scrollableEls2 = document.createElement('div');
+        scrollableEls2.className = 'scrollable-2';
+        scrollableEls.push(scrollableEls1, scrollableEls2);
+        document.body.appendChild(scrollableEls1);
+        document.body.appendChild(scrollableEls2);
+      });
+
+      afterEach(() => {
+        // Explicitly call `destroy` before removing scrollable elements
+        // from document to test the scrollable-remove-events path of
+        // the `destroy` method. Otherwise, this afterEach will execute
+        // before the test-global one that calls `destroy`, and there will
+        // be no scrollable elements left in the document.
+        bucketBar.destroy();
+        scrollableEls.forEach(el => el.remove());
+      });
+
+      it('should update buckets when any scrollable scrolls', () => {
+        bucketBar = createBucketBar({
+          scrollables: ['.scrollable-1', '.scrollable-2'],
+        });
+        assert.notCalled(fakeBucketUtil.buildBuckets);
+        scrollableEls[0].dispatchEvent(new Event('scroll'));
+        assert.calledOnce(fakeBucketUtil.buildBuckets);
+        scrollableEls[1].dispatchEvent(new Event('scroll'));
+        assert.calledTwice(fakeBucketUtil.buildBuckets);
+      });
+    });
+
+    it('should not update if another update is pending', () => {
```

Comment:
This is totally an internal implementation detail, but I was gunning for 100% test coverage ;)